src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -769,8 +798,13 @@ public Dimension getMinimumSize() {
     }
 
     /**
-     * The scrollbar is flexible along it's scrolling axis and
+     * The scrollbar is flexible along its scrolling axis and
      * rigid along the other axis.
+     * As specified in {@code setMaximumSize} JScrollBar will derive the
+     * maximum size from the preferred size in one axis and a
+     * fixed maximum size in the other.
+     *
+     * @return the maximum size as specified above.
```

Comment:
This can be amended in a similar way to `getMinimumSize`.

Returns the maximum size for the {@code JScrollBar}. The maximum size is equal to a fixed maximum value its scrolling axis and the preferred size along the other axis. This makes the scrollbar flexible along its scrolling axis and rigid along the other axis.